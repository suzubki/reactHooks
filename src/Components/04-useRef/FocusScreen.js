import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => { 
        document.querySelector('input').focus()
     }

  return (
    <div style={{padding: "70px"}} >
        <h1> FocusScreen </h1>
        <hr />

        <input 
            ref={inputRef}
            className='form-control'
            placeholder='Su nombre'
        />
        <br />

        <button 
            className='btn btn-primary'
            onClick={handleClick}
        >
            Focus
        </button >

    </div>
  )
}
