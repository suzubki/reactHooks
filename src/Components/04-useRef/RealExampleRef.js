import React, { useState } from "react";

import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {
    const [show, setShow] = useState(true);

    return (
        <div style={{ padding: "70px" }}>
            <h1>RealExampleRef</h1>
            <hr />

            {show && <MultipleCustomHooks />}

            <button className="btn btn-primary" onClick={() => setShow(!show)}>
                Show hide
            </button>
        </div>
    );
};
