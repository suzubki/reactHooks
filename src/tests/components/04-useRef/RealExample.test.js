import { RealExampleRef } from "../../../Components/04-useRef/RealExampleRef";
import { shallow } from "enzyme";

describe("RealExample", () => {
    const wrapper = shallow(<RealExampleRef />);
    test("component MultipleCustomHooks exists", () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
    });

    test("component MultipleCustomHooks no exists", () => {
        wrapper.find("button").simulate("click");
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
    });
});
