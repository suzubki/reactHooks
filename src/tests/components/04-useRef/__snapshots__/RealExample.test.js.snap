// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExample component MultipleCustomHooks exists 1`] = `
<div
  style={
    Object {
      "padding": "70px",
    }
  }
>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show hide
  </button>
</div>
`;
